### PR TITLE
Welcome page

### PR DESCRIPTION
**Pull request by quentinv72**
Created welcome page.
Added event listener to 'unmount' the scene on click event (for subject of choice).

### CODE REVIEW

File: src/components/roundCornerBox.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const RoundedBoxGeometry = require("three-rounded-box")(THREE);
+
+AFRAME.registerComponent("roundbox", {
+  schema: {
+    borderRadius: { type: "number", default: 0.15 },
+    opacity: { type: "number", default: 1 },
+  },
+  init: function () {
+    this.ascend = true;
+    const opacity = this.data.opacity;
```
Comment (amirhmk) [inline src/components/roundCornerBox.js:10]: Can we do object destruction here for clarity? 

`const { opacity, borderRadius } = this.data;`

File: src/components/roundCornerBox.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      new RoundedBoxGeometry(1, 1, 1, this.data.borderRadius, 50),
+      this.material
+    );
+    el.setObject3D("boxybox", myBox);
+    el.setAttribute("position", {
+      x: 3,
+      y: 1,
+      z: -3,
+    });
+    el.setAttribute("scale", { x: 1, y: 1, z: 1 });
```
Comment (amirhmk) [inline src/components/roundCornerBox.js:30]: Do we need this explicitly? Is it not the default case?

File: src/components/roundCornerBox.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    const el = this.el;
+    this.material = new THREE.MeshStandardMaterial({
+      opacity: opacity,
+      transparent: true,
+      color: 0xf7cb0d,
+      metalness: 0.1,
+      roughness: 0.2,
+    });
+    const myBox = new THREE.Mesh(
+      new RoundedBoxGeometry(1, 1, 1, this.data.borderRadius, 50),
```
Comment (amirhmk) [inline src/components/roundCornerBox.js:21]: Can use `borderRadius` if defined separately above

File: src/components/title.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ const fontJSON = require("../assets/Acme_Regular.json");
 const font = loader.parse(fontJSON);
 AFRAME.registerComponent("title", {
   init: function () {
+    const opacity = this.data.opacity;
     const geometry = new THREE.TextGeometry("Welcome to MathVR", {
```
Comment (amirhmk) [inline src/components/title.js:7]: Let's have this text as a variable that's passed in as well

File: src/components/title.js [hunk 5]
```diff
@@ -12,23 +13,18 @@ AFRAME.registerComponent("title", {
...
+      x: -5.75,
+      y: 6,
       z: -8,
     });
-    const el = this.el;
-    const sceneEl = el.sceneEl;
-    sceneEl.addEventListener("remove", function (event) {
-      sceneEl.remove(el);
-    });
+    this.el.setAttribute("scale", { x: 1, y: 1, z: 1 });
```
Comment (amirhmk) [inline src/components/title.js:27]: Don't need this

File: src/components/welcomeScreen.js [hunk 6]
```diff
@@ -1,40 +1,60 @@
...
       primitive: "cylinder",
       openEnded: true,
       thetaLength: 100,
       thetaStart: 180,
-      radius: 5,
-      height: 5,
+      radius: 10,
+      height: 10,
     });
+    this.el.setAttribute("scale", { x: 1, y: 1, z: 1 });
```
Comment (amirhmk) [inline src/components/welcomeScreen.js:13]: or this

File: src/components/welcomeScreen.js [hunk 7]
```diff
@@ -1,40 +1,60 @@
...
-      color: "green",
+      shader: "flat",
+      src: blackBoard,
     });
     this.el.setAttribute("position", {
       x: 0,
       y: 2,
       z: -5,
     });
+    this.el.setAttribute("rotation", {
```
Comment (amirhmk) [inline src/components/welcomeScreen.js:25]: This doesn't seem to do anything as the one block below overwrites it?

File: package.json [hunk 8]
```diff
@@ -27,7 +27,8 @@
     "diff-dom": "^4.1.6",
     "html-require-loader": "^1.0.1",
     "lodash": "^4.17.15",
-    "moment": "^2.25.3"
+    "moment": "^2.25.3",
```
[the remark this answers is not in the document]
Comment (quentinv72) [reply package.json:30]: This has been resolved

File: src/components/roundCornerBox.js (hunk 1 above)
Comment (quentinv72) [reply src/components/roundCornerBox.js:10]: Resolved

File: src/components/fadeaway.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+AFRAME.registerComponent("fadeaway", {
+  init: function () {
+    const sceneEl = this.el.sceneEl;
+    const el = this.el;
+    sceneEl.addEventListener("fadeaway", function (event) {
+      const elements = sceneEl.querySelectorAll(`.${event.detail.className}`);
+      console.log(elements);
+      for (let element of elements) {
+        console.log(elements);
```
Comment (amirhmk) [inline src/components/fadeaway.js:9]: stray `console.log`

File: src/components/roundCornerBox.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      this.material
+    );
+    el.setObject3D("boxybox", myBox);
+    el.setAttribute("position", {
+      x: 3,
+      y: 1,
+      z: -3,
+    });
+  },
+  // tick: function () {
```
Comment (amirhmk) [inline src/components/roundCornerBox.js:29]: Do we need this?

**Pull request merged**